interpreter.asm

```diff
@@ -23,16 +23,47 @@
 ; QUIT INTERPRET FIND FIND-NAME >CFA PARSE-NAME WORD EXECUTE EVALUATE ' ABORT /STRING
 
 restore_handler
-    cli
-    jmp QUIT
-
-quit_reset
+   pha				; save a
+   txa				; copy x
+   pha				; save x
+   tya				; copy y
+   pha				; save y
+   lda	#$7f	    ; disable all interrupts
+   sta	$dd0d       ; 
+   ldy	$dd0d       ; save cia 2 icr
+   bpl stop_restore ; if high bit not set
+
+kernal_nmi
+   jmp $fe72
+
+stop_restore
+    jsr	$f6bc		; increment real time clock
+                    ; scan stop key 
+   	lda	$91 		; read the stop key column
+	cmp	#$7f		; compare with [stp] down
+	        		; if not [stp] or not just [stp] exit
+    bne	kernal_nmi	; if not [stop] restore registers and exit interrupt
+    
+brk_handler
+   pla
+   pla
+   tax              ; restore xr for QUIT
```

Comment:
Does this pla/pla/tax properly restore x even for BRK command? How does that work?

Reply:
...looking at how BRK works, I guess x would be overwritten with some seemingly random value?
If so, I think it could be non-controversial to not touch $316 at all. It saves a few bytes and it does not really take away any possibilities.

Reply:
Y is at the top of the return stack at this point, and X is underneath. QUIT then resets the return stack preserving X.
  It was never certain that X would be valid, if Restore was initiated during IRQ or while assembly has X stored away, but it usually was good.

Reply:
OK cool! Yes I haven't really run this code. So if you say it works, it's fine for me.

Reply:
> ...looking at how BRK works, I guess x would be overwritten with some seemingly random value?
> If so, I think it could be non-controversial to not touch $316 at all. It saves a few bytes and it does not really take away any possibilities.

 BRK arrives at the vector with the registers intact, return address and status on the return stack. 
  So if you insert a [ brk, } into your forth code, it'll preserve the data stack. I think this is useful.
  Also, making run/stop restore and BRK equivalent emulates the kernals' handler, and makes BRK safe.